website/pages/blog/type-safe-your-api.mdx

```diff
@@ -0,0 +1,132 @@
+
+## How I Used Web API before feTS?
+
+In this blog post, we will use the Spotify API as an example, but you can apply these principles to any REST API of your choice. You can find the complete code example [here](https://github.com/TuvalSimha/feTS-example).
+
+Spotify provides a developer-friendly API that is easy to use. Here's a example of how to retrieve your top 5 tracks from the last 30 days using the Spotify API:
+
+1. Create a Spotify account and retrieve your access token from [here](https://developer.spotify.com/documentation/web-api/reference/get-users-top-artists-and-tracks?type=artists&time_range=short_term&limit=5&offset=0).
+2. Create a fetchWebApi function that fetches data from the API and returns it as JSON:
+    
+```typescript
+const token = 'YOUR_TOKEN_HERE';
+
+async function fetchWebApi(endpoint: string, method: string, body: undefined) {
+  const res = await fetch(`https://api.spotify.com/${endpoint}`, {
+    headers: {
+      Authorization: `Bearer ${token}`,
+    },
+    method,
+    body: JSON.stringify(body),
+  });
+  
+  return await res.json();
+}    
+```
+
+3. Call the fetchWebApi function with the desired endpoint:
+```typescript
+async function getTopTracks() {
+  // Endpoint reference: https://developer.spotify.com/documentation/web-api/reference/get-users-top-artists-and-tracks
+  return (await fetchWebApi(
+    'v1/me/top/tracks?time_range=short_term&limit=5',
+    'GET',
+    undefined
+  )).items;
+}
+```
+4. Lastly, you can use the data from the API:
+```typescript
+async function main() {
+  const topTracks = await getTopTracks();
+  console.log(topTracks);
+}
+
+main();
+```
+
+While this approach works, it has some drawbacks:
+
+- The `fetchWebApi` function lacks type safety, making it uncertain what data will return from the API.
+- You need to specify the parameters of the `fetchWebApi` function manually.
+- The `fetchWebApi` function allows calling any endpoint, even if it's not supported by the API.
+- The `topTracks` variable lacks type safety according to the API documentation.
+
+## How I Use Web API with feTS?
+To use the Spotify API with feTS, follow these steps:
+
+1. Create a new TypeScript project and install feTS from [here](https://the-guild.dev/openapi/fets/client/quick-start).
+
+2. Create a TypeScript file that exports your OpenAPI document as a constant:
+```typescript filename="openapi.ts"
+export default { openapi: '3.0.0' /* ... */ } as const;
+```
+In our case, we will use the Spotify API, so we can retrieve the OpenAPI document from [here](https://developer.spotify.com/reference/web-api/open-api-schema.yaml). Since this OpenAPI document is in YAML format, we need to convert it to JSON format before using it in our project. You can find an example of how to perform the conversion [here](https://github.com/TuvalSimha/feTS-example/blob/main/convert-yaml-to-json.ts).
```

Comment:
You can link this — https://api.apis.guru/v2/specs/spotify.com/sonallux/2023.2.27/openapi.json — it's already in the JSON format